src/core/ledger/ledger.js

```diff
@@ -768,6 +774,80 @@ export class Ledger {
     account.payoutAddresses.delete(JSON.stringify(currencyResult.value));
   }
 
+  /**
+   * no-ops if integration tracking is already enabled
+   */
+  enableIntegrationTracking(): Ledger {
+    if (!this._trackGrainIntegration) {
+      this._createAndProcessEvent({
+        type: "ENABLE_GRAIN_INTEGRATION",
+      });
+    }
+    return this;
+  }
+
+  /**
+   * no-ops if integration tracking is already disabled
+   */
+  disableIntegrationTracking(): Ledger {
+    if (this._trackGrainIntegration) {
+      this._createAndProcessEvent({
+        type: "DISABLE_GRAIN_INTEGRATION",
+      });
+    }
+    return this;
+  }
+  _toggleGrainIntegrationTracking(_: ToggleGrainIntegration) {
+    this._trackGrainIntegration = !this._trackGrainIntegration;
+    if (!this._trackGrainIntegration)
+      this._grainIntegrationDistributions.clear();
+  }
```

Comment:
nit: Not totally sold on this. Seems like an unnecessary addition of complexity / point of failure / deviation from the pattern.